src/serlio/prtModifier/MayaCallbacks.cpp

```diff
@@ -251,13 +251,12 @@ void MayaCallbacks::addMesh(
 			if (size > 0) {
 				for (unsigned int i=0; i<arrayLength; i++) {
 					size_t maxStringLengthTmp = maxStringLength;
-					char* tmp = new char[maxStringLength];
+					std::array<char, maxStringLength> tmp;
 					std::wstring keyToUse = key;
 					if (i>0)
 						keyToUse = key + std::to_wstring(i);
-					prt::StringUtils::toOSNarrowFromUTF16(keyToUse.c_str(), tmp, &maxStringLengthTmp);
-					fStructure->addMember(type, size, tmp);
-					delete tmp;
+					prt::StringUtils::toOSNarrowFromUTF16(keyToUse.c_str(), tmp.data(), &maxStringLengthTmp);
+					fStructure->addMember(type, size, tmp.data());

```

Comment:
'type' is not initialized for all cases

Reply:
`type` is always initialized, since we are in a block that is guarded by `size > 0` and `size` is only set to something greater than 0 iff we also set the type.

Although I agree, the code structure around this part is far from ideal.